### DISCUSSION TITLE
Problems with table method in the listing

### DISCUSSION THREAD

**Discussion by thiagoalgo** [filamentphp/filament#12130] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.61

### How can we help you?

In the documentation found at https://filamentphp.com/docs/3.x/panels/resources/listing-records#customizing-the-table-eloquent-query the table method signature is static. However, when trying to override this static method it gives an error stating that it is not possible. As I'm using PHPStorm, it suggests signing the method as non-static.

Using the method as non-static, I am modifying the query according to the code below:

```php
    public function table(Table $table): Table
    {
        return $table
            ->modifyQueryUsing(fn (Builder $query) => $query->where('id', '<>', 0));

    }
```

Using the DebugBar I can see that the query is being modified, but the table is not displayed as shown in the image below:

![image](https://github.com/filamentphp/filament/assets/488512/f60dca72-caee-43f8-bd14-cd5a819f5542)

![image](https://github.com/filamentphp/filament/assets/488512/24010b27-de80-4be1-addf-5209bcfc1c8f)

Does anyone know what could be happening?

**Comment by leandrocfe** [comment, marked as answer]
This static method is available in the Resource.php file, for example

```php
//app/Filament/Resources/UserResource.php
public static function table(Table $table): Table
{
    return $table
        ->modifyQueryUsing(fn (Builder $query) => $query->where('id', '<>', 0))
...
```

**Reply by thiagoalgo** [reply]
Perfect, Leandro. Understood. Although I think the documentation is a bit confusing, as I am on the topic of listing records and the doc suggests that it is in the listing class, according to the excerpt below:

"Although you can customize the Eloquent query for the entire resource, you can also make specific modifications for the List page table."

Anyway, question answered. Thank you very much.
